osu.Desktop/OsuGameDesktop.cs

```diff
@@ -125,7 +126,10 @@ public override bool RestartAppWhenExited()
                     return true;
             }
 
-            return base.RestartAppWhenExited();
+            SessionStatics.SetValue(Static.RestartRequested, true);
+            AttemptExit();
```

Comment:
This is never reached in Windows because of the early return above.

Reply:
This seems valid to me, it looks like the game would not actually exit itself.